Shared/Samples/Filter by definition expression or display filter/FilterByDefinitionExpressionOrDisplayFilterView.swift

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct FilterByDefinitionExpressionOrDisplayFilterView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The draw status of the map view.
+    @State private var drawStatus: DrawStatus?
+    
+    /// The result of counting the features in the current extent of the map view.
+    @State private var featureCountResult: Result<Int, Error>?
+    
+    /// The filtering mode selected in the picker.
+    @State private var selectedFilterMode: FilterMode = .none
+    
+    var body: some View {
+        GeometryReader { geometryProxy in
+            MapViewReader { mapViewProxy in
+                MapView(map: model.map)
+                    .onDrawStatusChanged { drawStatus = $0 }
+                    .task(id: drawStatus) {
+                        // Updates the feature count when the map view finishes drawing.
+                        guard drawStatus == .completed else { return }
+                        
+                        // Creates an envelope from the frame of the map view.
+                        let viewRect = geometryProxy.frame(in: .local)
+                        
+                        guard let viewExtent = mapViewProxy.envelope(fromViewRect: viewRect) else {
+                            return
+                        }
+                        
+                        featureCountResult = await Result {
+                            // Gets the feature count contained within the envelope.
+                            try await model.numberOfFeatures(withinExtent: viewExtent)
+                        }
```

Comment:
`Result.init(awaiting:)` is provided by the toolkit and, in my opinion, [it really shouldn't be](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/732). Consider not using it.

```suggestion
                        do {
                            // Gets the feature count contained within the envelope.
                            let count = try await model.numberOfFeatures(withinExtent: viewExtent)
                            featureCountResult = .success(count)
                        } catch {
                            featureCountResult = .failure(error)
                        }
```